rust/lance-datafusion/src/exec.rs

```diff
@@ -240,17 +244,17 @@ lazy_static! {
 }
 
 pub fn get_session_context(options: &LanceExecutionOptions) -> SessionContext {
-    let session_ctx: SessionContext;
+    if options.target_partition.is_some() {
+        return new_session_context(options);
+    }
     if options.mem_pool_size() == DEFAULT_LANCE_MEM_POOL_SIZE {
```

Comment:
Minor nit: the changes to this method could be simplified if you changed this line to:

```
if options.mem_pool_size() == DEFAULT_LANCE_MEM_POOL_SIZE && options.target_partition.is_none() {
```